jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java

```diff
@@ -36,18 +36,17 @@
   }
 
   /**
-   * Create DMatrix from iterator.
-   *
-   * @param iter The data iterator of mini batch to provide the data.
-   * @param cacheInfo Cache path information, used for external memory setting, can be null.
-   * @throws XGBoostError
-   */
-  public DMatrix(Iterator<LabeledPoint> iter, String cacheInfo) throws XGBoostError {
+     * Create DMatrix from iterator.
+     *
+     * @param iter The data iterator of mini batch to provide the data.
+     * @param cacheInfo Cache path information, used for external memory setting, can be null.
+     * @param batchSize Size of batch, recommended 32 << 10
+     * @throws XGBoostError
+     */
+  public DMatrix(Iterator<LabeledPoint> iter, String cacheInfo, int batchSize) throws XGBoostError {
     if (iter == null) {
       throw new NullPointerException("iter: null");
     }
-    // 32k as batch size
-    int batchSize = 32 << 10;
```

Comment:
This means if users call XGBoost Java API to train a model, then there is no default batch size; but there is default value for Scala API. Is this by design? I think here we should follow the Scala way to set default value in arguments.

Reply:
added constructor with default batch size